### ISSUE TITLE
`exit <code>` not working correctly anymore (after mvdan.cc/sh upgrade?)

### ISSUE THREAD

**Issue by antonmarin** [go-task/task#251]
Status broken or misbehavior with docs:

> Alternatively, you can inform a sequence of tests as status. If no error is returned (exit status 0), the task is considered up-to-date:

- Task version 2.6.0
- OS macOS High Sierra
- Example Taskfile showing the issue

```
  asd:
    cmds:
      - echo test
    status:
      - exit 0
```
result
```
▶ task asd --status -v
task: dynamic variable: 'uname | tr '[:upper:]' '[:lower:]'' result: 'darwin'
task: dynamic variable: 'echo "development"' result: 'development'
task: status command exit 0 exited non-zero: exit status 0
task: Task "asd" is not up-to-date

▶ task asd -v         
task: dynamic variable: 'uname | tr '[:upper:]' '[:lower:]'' result: 'darwin'
task: dynamic variable: 'echo "development"' result: 'development'
task: status command exit 0 exited non-zero: exit status 0
echo test
test
```

**Comment by andreynering** [comment]
Hi @antonmarin,

I noted that `status` is actually working fine, but there's a bug on calling `exit` directly in commands or status.

**Comment by andreynering** [comment]
Hi @antonmarin,

This is now fixed on master.

Let me know if you're still facing problems.

**Comment by antonmarin** [comment]
Great! Going to try soon

**Comment by antonmarin** [comment]
Are you going to include this fix to release?

**Comment by andreynering** [comment]
@antonmarin Just released as v2.7.1
